### ISSUE TITLE
anchor: almost all relative references will be broken

### ISSUE THREAD

**Issue by LionNatsu** [AOSC-Dev/pug-zh-cn#2]
Please use
```
grep -re '.html#'
```
to check them out.

Anchors are generated automatically in `h2`, `h3`, `h4`... 
*`h1` tags are not be affected.*

And now we should use
```
<h2 id="removed-language-features">已移除的语言特性</h2>
```
instead of
```
## 已移除的语言特性
```

An offical example: http://daringfireball.net/projects/markdown/syntax.text

**Comment by Artoria2e5** [comment]
@LionNatsu > instead of ##

In Wikipedia people sometimes use an empty span element with an id to create "compatibility anchors".

**Comment by LionNatsu** [comment]
To know what the anchor excatly is, check it out here https://pugjs.org/

**Comment by LionNatsu** [comment]
> In Wikipedia people sometimes use an empty span element with an id to create "compatibility anchors".

Yes, that's better. @liushuyu let's do so.

**Comment by LionNatsu** [comment]
Fixed at 3739c03f82c10ea6f3f7c283bfef41f40265705b.
